### PR TITLE
Aligned chip-types.xml with the latest Matter spec

### LINKED ISSUE #28976: [XML] Align chip-types.xml with the spec while making a nod towards the data model number allocation
IDs for data types are not used in Matter over the wire, so there is no impact to interop, and the SDK can readily be changed.  Align the https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/data-model/chip/chip-types.xml with the Matter spec in line with changes in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7054

### PR DESCRIPTION
**Pull request by robszewczyk**
* reordered the XML to use the same order as the spec; this should make maintenance a bit more sane
* renumbered types to avoid duplication and conflicts and to align with the spec

Fixes #28976

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]
We should probably add a lint that there are no duplicate IDs in that XML...

File: src/app/zap-templates/zcl/data-model/chip/chip-types.xml [hunk 1]
```diff
@@ -57,39 +55,57 @@ limitations under the License.
...
+    <type id="0xC7" description="Data Version"             name="data_ver"          size="4"  discrete="true" />
+    <type id="0xC8" description="Event Number"             name="event_no"          size="8"  discrete="true" />
+
+    <type id="0xF2" description="IP Address"               name="ipadr"                       composite="true"/>
+    <type id="0xF3" description="IPv4 Address"             name="ipv4adr"           size="4"  composite="true"/>
+    <type id="0xF4" description="IPv6 Address"             name="ipv6adr"           size="16" composite="true"/>
+    <type id="0xF5" description="IPv6 Prefix"              name="ipv6pre"                     composite="true"/>
+    <type id="0xF6" description="Hardware Address"         name="hwadr"                       composite="true"/>
+
+    <type id="0xC9" description="Semantic Tag"             name="semtag"            size="4"  discrete="true" />
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/chip-types.xml:105]: Could you add the new types to https://github.com/project-chip/connectedhomeip/blob/master/scripts/py_matter_idl/matter_idl/generators/types.py#L165 as well?

**Review by andy31415 — APPROVED** [review]

**Comment by robszewczyk** [comment]
> We should probably add a lint that there are no duplicate IDs in that XML...

Added a unique constraint to the XSD, and chip-types.xml to the validate.sh script; verified it fails on duplicate IDs

File: src/app/zap-templates/zcl/data-model/chip/chip-types.xml (hunk 1 above)
Comment (robszewczyk) [reply src/app/zap-templates/zcl/data-model/chip/chip-types.xml:105]: done.

**Pull request merged**